target/linux/ath79/dts/ar9132_buffalo_wzr-hp-g300nh.dts

```diff
@@ -0,0 +1,259 @@
+		poll-interval = <20>;
+
+		aoss {
+			label = "aoss";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
+		router_on {
+			label = "router_on";
+			linux,code = <BTN_5>;
+			gpios = <&gpio2 2 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		qos_off {
+			label = "qos_off";
+			linux,code = <BTN_3>;
+			gpios = <&gpio2 3 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+*/
+		usb {
+			label = "usb";
+			linux,code = <BTN_2>;
+			gpios = <&gpio2 5 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
+		router_auto {
+			label = "router_auto";
+			linux,code = <BTN_6>;
+			gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		qos_on {
+			label = "qos_on";
+			linux,code = <BTN_4>;
+			gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+*/
+	};
+
+	flash@1e000000 {
+		compatible = "cfi-flash";
+		reg = <0x1e000000 0x2000000>;
+		bank-width = <2>;
+		device-width = <2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0000000 0x0040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x0040000 0x0020000>;
+				read-only;
+			};
+
+			partition@60000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x0060000 0x1f60000>;
+			};
+
+			partition@1fc0000 {
+				label = "user_property";
+				reg = <0x1fc0000 0x0020000>;
+				read-only;
+			};
+
+			art: partition@1fe0000 {
+				label = "art";
+				reg = <0x1fe0000 0x020000>;
+				read-only;
+			};
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		usb {
+			label = "buffalo:blue:usb";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port>;
+			linux,default-trigger = "usbport";
+		};
+
+		led_diag: diag {
+			label = "buffalo:red:diag";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		wireless {
+			label = "buffalo:green:wireless";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_security: security {
+			label = "buffalo:amber:security";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+
+		router {
+			label = "buffalo:green:router";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	switch {
+		compatible = "realtek,rtl8366s";
+		gpio-sda = <&gpio 19 GPIO_ACTIVE_HIGH>;
+		gpio-sck = <&gpio 20 GPIO_ACTIVE_HIGH>;
+		mii-bus = <&mdio0>;
+
+		mdio-bus {
+			status = "okay";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+			phy-mask = <0x10>;
+
+			phy4: ethernet-phy@4 {
+				reg = <4>;
+				phy-mode = "rgmii";
+			};
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	pll-data = <0x1e000100 0x13000a44 0x00441099>;
+	phy-mode = "rgmii";
+	mtd-mac-address = <&art 0x1120c>;
```

Comment:
Can you check whether there are additional valid addresses in u-boot or art partition?
I'd particularly be interested in art 0x10000 and 0x10006 ...

Reply:
How do I recognise a valid address if I see one? :)

Reply:
The MAC assignment should mirror that from OEM firmware. So, if Buffalo uses the same MAC address on all interface, please do the same for OpenWrt.

If you find more addresses, just report them here (location and MAC address, the last byte of the address should be sufficient). Then we will see.

A "valid" address will be one that's either the address from OEM firmware or a close increment of it. An easy way is to just hexdump -C the whole partition and then grep for parts of the MAC address (note that you won't match cases wrapped around the middle or the end of one line if you try to match more than one byte).

Further reading: https://openwrt.org/docs/guide-developer/mac.address

Reply:
> Can you check whether there are additional valid addresses in u-boot or art partition?
> I'd particularly be interested in art 0x10000 and 0x10006 ...

It's all 'ff' up until 0x11200 on art and it does not seem to contain any more addresses. U-boot does not seem to contain any addresses at all.

Reply:
okay, thanks for looking. Then let's keep this for eth.